### PR TITLE
[EN-64448] - Dalia/geo tests

### PR DESCRIPTION
**Pull request by dnabilali**
Tests have been added to cover the following examples of California State Parks use cases (communicated through Stephen Z.):
 - Map of how many customers based on gender (male/female/nonbinary/etc) made reservations within a certain zip code, county, electoral district
 - Map of how many customers based on classifcations (veteran/differently abled/special pass/etx) made reservations within a certain zip code, county, electoral district
 - Numeric presentation - showing the average distance a customer traveled to get to that park (column name is 'place') based on the customer's zip code

### CODE REVIEW

**Review by zachlefevre — APPROVED** [review]

File: common-redshift/src/test/scala/com/socrata/common/sqlizer/SoQLSqlizerTest.scala [hunk 1]
```diff
@@ -1730,4 +1727,49 @@ class SoQLSqlizerTest {
       "(SELECT x6.i1 AS i5, x6.i2 AS i6 FROM ((SELECT x2.text AS i1, x2.num AS i2 FROM table1 AS x2) UNION (SELECT x4.text AS i3, x4.num AS i4 FROM table2 AS x4)) AS x6) UNION ALL (SELECT x8.text AS i7, x8.num AS i8 FROM table3 AS x8)"
     )
   }
+
+  @Test
+  def `group by and where clauses`(): Unit = {
+    /*Map of how many customers based on gender (male/female/nonbinary/etc) made reservations within a certain zip code, county, electoral district*/
+    assertEquals(
+      analyzeStatement("Select count(text), gender where substring(zip_code, 1, 5) = '12345' group by gender"),
+      "SELECT (count(x1.text)) :: decimal(30, 7) AS i1, x1.gender AS i2 FROM table1 AS x1 WHERE (substring(x1.zip_code, (1 :: decimal(30, 7)) :: int, (5 :: decimal(30, 7)) :: int)) = (text '12345') GROUP BY x1.gender"
```
Comment (ds2643) [inline common-redshift/src/test/scala/com/socrata/common/sqlizer/SoQLSqlizerTest.scala:1736]: What's the meaning of this subexpression?
```
(substring(x1.zip_code, (1 :: decimal(30, 7)) :: int, (5 :: decimal(30, 7)) :: int))
```

Comment (dnabilali) [reply common-redshift/src/test/scala/com/socrata/common/sqlizer/SoQLSqlizerTest.scala:1736]: it creates a substring of length 5 characters starting from the first character:
SUBSTRING(character_string, start_position, number_characters ).

**Comment by zachlefevre** [comment]
@dnabilali I think you should make substring *not* cast to "int", but instead cast to `decimal(30, 7)`.

**Comment by zachlefevre** [comment]
I think we should make all of these `int` `decimal(30, 7)`
```
      RightPad -> sqlizeNormalOrdinaryFuncall(
        "rpad",
        castType = idx =>
          idx match {
            case 1 => Some(Doc("int"))
            case _ => None
          }
      ),
      Chr -> sqlizeNormalOrdinaryFuncall(
        "chr",
        castType = idx =>
          idx match {
            case 0 => Some(Doc("int"))
            case _ => None
          }
      ),
      Substr2 -> sqlizeNormalOrdinaryFuncall(
        "substring",
        castType = idx =>
          idx match {
            case 1 => Some(Doc("int"))
            case _ => None
          }
      ),
      Substr3 -> sqlizeNormalOrdinaryFuncall(
        "substring",
        castType = idx =>
          idx match {
            case 1 => Some(Doc("int"))
            case 2 => Some(Doc("int"))
            case _ => None
          }
      ),
      SplitPart -> sqlizeNormalOrdinaryFuncall(
        "split_part",
        castType = idx =>
          idx match {
            case 2 => Some(Doc("int"))
            case _ => None
          }
      )
      ```

**Comment by dnabilali** [comment]
will merge now after verifying with @zachlefevre that `int` is the correct datatype casting for the above functionalities in redshift.

**Pull request merged**